smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -281,4 +305,43 @@ public Void serviceShape(ServiceShape shape) {
 
         return null;
     }
+
+    private boolean shapesMatch(Shape shape, Shape otherShape) {
+        // Check names match.
+        if (!shape.getId().getName().equals(otherShape.getId().getName())) {
+            return false;
+        }
+
+        // Check traits match.
+        Map<ShapeId, Trait> traits = shape.getAllTraits();
+        Map<ShapeId, Trait> otherTraits = otherShape.getAllTraits();
+        for (ShapeId trait : traits.keySet()) {
+            // Ignore box trait when comparing.
+            if (!otherTraits.containsKey(trait) && !trait.getName().equals("box")) {
+                return false;
+            }
+        }
+        for (ShapeId trait : otherTraits.keySet()) {
+            // Ignore box trait when comparing.
+            if (!traits.containsKey(trait) && !trait.getName().equals("box")) {
+                return false;
+            }
+        }
+
+        // Check members match.
```

Comment:
This also looks like it could just be `shape.members().equals(otherShape.members())`

Reply:
Doing the check with `shape.members().equals(otherShape.members())` way won't ignore the namespaces:

```
namespace smithy.example

structure Bar {
    @required
    baz: String
}

namespace another.example

structure Bar {
    @required
    baz: String
}
```

The `baz` members are different, so it returns `false`.

Reply:
:+1: